test/login-form.test.tsx

```diff
@@ -0,0 +1,61 @@
+import { mount, shallow } from 'enzyme'
+import React from 'react'
+import TextField from '@material-ui/core/TextField'
+import { act } from 'react-dom/test-utils'
+import { LoginForm } from '../src/components/login-form'
+
+describe('LoginForm', () => {
+  const testprop = {
+    onLogin: jest.fn(),
+    error: 'test',
+  }
+
+  it('LoginForm snapshot', () => {
+    const wrapper = shallow(<LoginForm {...testprop} />)
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  it('Username change test', async () => {
+    const wrapper = mount(<LoginForm {...testprop} />)
+    act(() => {
+      ;(wrapper
+        .find(TextField)
+        .first()
+        .prop('onChange') as any)({ target: { value: 'testusername' } } as any)
+    })
+    expect(
+      wrapper
+        .update()
+        .find(TextField)
+        .first()
+        .props().value,
+    ).toEqual('testusername')
+  })
+
+  it('Password change test', async () => {
+    const wrapper = mount(<LoginForm {...testprop} />)
+    act(() => {
+      ;(wrapper
+        .find(TextField)
+        .at(1)
+        .prop('onChange') as any)({ target: { value: 'testpassword' } } as any)
+    })
```

Comment:
What is the expectation here?

Reply:
line 20-25 is a change event on the Textfield. I pass the { target: { value: 'testusername' }} object. Then I expect the Textfield value. Did you think of this?

Reply:
Oh... I left it there. I will remove it